include/utility.hpp

```diff
@@ -465,7 +465,7 @@ template <class SH_CORE> class TShardsVar {
         auto msg = "TShardsVar: Error during blocks cleanup: " + std::string(errors.message);
         SH_CORE::log(msg.c_str());
       }
-      blk->destroy(blk);
+      decRef(blk);
```

Comment:
Not super happy about this but I guess...

Reply:
I wanted to have the same behavior everywhere to avoid issues in the feature.
So it's either all or no reference counting